### PR TITLE
James 2393 integration test

### PR DESCRIPTION
**Pull request by mbaechler**
That's a rework of https://github.com/linagora/james-project/pull/1407 where I rebased on https://github.com/linagora/james-project/pull/1415 and dropped some injection code to prepare configuration loading (https://ci.linagora.com/linagora/lgs/openpaas/james/issues/622)

of course, the first commit should be skipped during review

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/QuotaMailingTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    @Test
+    public void shouldSendANoticeWhenThresholdExceeded() throws Exception {
+        jmapServer.getProbe(QuotaProbesImpl.class)
+            .setMaxStorage(MailboxConstants.USER_NAMESPACE + "&" + HOMER,
+                new SerializableQuotaValue<>(QuotaSize.size(100 * 1000)));
+
+        bartSendMessageToHomer();
+        // Home receives a 83.995 Bytes mail, triggering the 0.80 threshold mailing
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/QuotaMailingTest.java]: *Homer

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdMailer.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import java.io.IOException;
 import java.util.Optional;
 
+import javax.inject.Inject;
```
Comment (rouazana) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdMailer.java]: unused import?

Comment (chibenwa) [reply mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdMailer.java]: Outdated

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[caa0dc1a12ff9213a700764bd893a45ea91cb0e0] Tests in error: 
[caa0dc1a12ff9213a700764bd893a45ea91cb0e0]   GroupMappingTest.messageShouldRedirectToUserContainingSlash:472 » ConditionTimeout
```

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
```
[39562a1a6f71287711a10973831e523e872a6116] Tests in error: 
[39562a1a6f71287711a10973831e523e872a6116]   ToRepositoryTest.userShouldBeAbleToAccessReprocessedMails:130 » ConditionTimeout
```

test this please

**Comment by chibenwa** [comment]
```
[39562a1a6f71287711a10973831e523e872a6116] Failed tests: 
[39562a1a6f71287711a10973831e523e872a6116]   ManageableMailQueueContract.concurrentEnqueueShouldNotAlterBrowsing:245 
[39562a1a6f71287711a10973831e523e872a6116] Actual and expected should have same size but actual size was:
[39562a1a6f71287711a10973831e523e872a6116]   <3>
[39562a1a6f71287711a10973831e523e872a6116] while expected size was:
[39562a1a6f71287711a10973831e523e872a6116]   <2>
[39562a1a6f71287711a10973831e523e872a6116] Actual was:
[39562a1a6f71287711a10973831e523e872a6116]   <["name1", "name2", "name3"]>
[39562a1a6f71287711a10973831e523e872a6116] Expected was:
[39562a1a6f71287711a10973831e523e872a6116]   <["name1", "name2"]>
```

test this please

(ActiveMQ test failure)

**Comment by chibenwa** [comment]
```
[39562a1a6f71287711a10973831e523e872a6116] Feature: Sharing children mailbox.Scenario: User can share sub-mailbox without sharing its parent and then sharee can see the parent mailbox
[39562a1a6f71287711a10973831e523e872a6116]   Run 1: CassandraMailboxSharingTest » Configuration Guice configuration errors:
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] 1) Co...
[39562a1a6f71287711a10973831e523e872a6116]   Run 2: CassandraMailboxSharingTest » Configuration Guice configuration errors:
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] 1) Co...
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] Tests run: 2599, Failures: 0, Errors: 205, Skipped: 1999
```

**Comment by chibenwa** [comment]
Works locally.

test this please

**Comment by chibenwa** [comment]
```
[39562a1a6f71287711a10973831e523e872a6116] Feature: Sharing children mailbox.Scenario: User can share sub-mailbox without sharing its parent and then sharee can see the parent mailbox
[39562a1a6f71287711a10973831e523e872a6116]   Run 1: CassandraMailboxSharingTest » Configuration Guice configuration errors:
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] 1) Co...
[39562a1a6f71287711a10973831e523e872a6116]   Run 2: CassandraMailboxSharingTest » Configuration Guice configuration errors:
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] 1) Co...
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] 
[39562a1a6f71287711a10973831e523e872a6116] Tests run: 2599, Failures: 0, Errors: 205, Skipped: 1999
```

Again...

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**